src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -456,116 +525,129 @@ private async Task AcceptStreamsAsync()
         private async Task ProcessServerStreamAsync(QuicStream stream)
         {
             ArrayBuffer buffer = default;
+            QuicStream? quicStream = stream;
 
             try
             {
-                await using (stream.ConfigureAwait(false))
+                buffer = new ArrayBuffer(initialSize: 32, usePool: true);
+                int bytesRead;
+
+                try
                 {
-                    if (stream.CanWrite)
-                    {
-                        // Server initiated bidirectional streams are either push streams or extensions, and we support neither.
-                        throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
-                    }
+                    // webtransport has a header of 3 bytes and in order to avoid reading
```

Comment:
Yes, but the stream type is always 2 bytes and the quic stream id will always be 1 byte. Or at least there would be **at least** 3 bytes, but I can't read more than that because I risk entering the user content inside the stream

Reply:
What will happen to the code in this method if at some point a new stream type will be added which is encoded in 8 bytes?

Reply:
@CarnaViire  Since `TryRead` is a bool function we can check for failure  of reading streamType at line 518 and we can read afterwards more bytes. I will implement this fix

Reply:
Fixed it, added more reeding in case ok bigger stream types

Reply:
> Since TryRead is a bool function we can check for failure of reading streamType at line 518 and we can read afterwards more bytes.

This is correct 👍

Can you find a place where a similar problem was already solved (reading first integer(s) and not reading anything else)?

Reply:
Inside `TryReadIntegerPair` we try to read the first 2 integers but if we fail we return false